docs/slack-spring-boot-docs/src/main/asciidoc/dev-features/kotlin-sample-extensions.adoc

```diff
@@ -1,11 +1,152 @@
 [[kotlin-sample-extensions]]
 = Kotlin Sample Extensions
 
-We also built an extension package which extends all dto´s with .sample() methods, you can use these methods to create sample-objects for testing.
+This guide will guide you through the process of building your own Slash Command Receiver class with Slack Spring Boot Starter.
+We will use the extension package which extends all DTOs with .sample() methods, which will come in handy for testing.
+
+[NOTE]
+====
+Slack Spring Boot Starter only supports Kotlin
+====
+
+=== What you will need
+
+- An IDE of your choice - preferably IntelliJ IDEA (Community Edition will just be fine)
+- at least JDK 1.8
+- Gradle 5
+
+=== Preparation
+
+First you'll need to have a build.gradle file
+
+To use the test-extensions, simply include the following in your build.gradle:
 
-== Install
-to use the test-extensions, simply include the following in your build.gradle:
 [source]
 --
 testImplementation(project(":slack-jackson-dto-test"))
---
\ No newline at end of file
+--
+
+Additionally we want to have have
+
+[source]
+--
+implementation(project(":slack-spring-test-api-client"))
+--
+
+in our build.gradle in order to test the functionality of our example.
+
+=== The Slash Command Receiver
+
+The https://github.com/kreait/slack-spring-boot-starter/blob/master/starter/slack-spring-boot/src/main/kotlin/com/kreait/slack/broker/receiver/SlashCommandReceiver.kt[Slash Command Receiver] is an interface that is based around handling Slash Commands.
+
+Let's assume we have the following class that implements the https://github.com/kreait/slack-spring-boot-starter/blob/master/starter/slack-spring-boot/src/main/kotlin/com/kreait/slack/broker/receiver/SlashCommandReceiver.kt[Slash Command Receiver]:
+
+[source,kotlin]
+--
+include::../../../../../../samples/slack-spring-boot-starter-sample/src/main/kotlin/com/kreait/slack/sample/ResponseCommandReceiver.kt[]
+--
+
+We provide a https://github.com/kreait/slack-spring-boot-starter/blob/master/client/slack-api-client/src/main/kotlin/com/kreait/slack/api/SlackClient.kt[SlackClient] and a https://github.com/kreait/slack-spring-boot-starter/blob/master/samples/slack-spring-boot-starter-sample/src/main/kotlin/com/kreait/slack/sample/ResponseHandler.kt[ResponseHandler] in the constructor which are essential for the functionality of our class.
```

Comment:
I would say that we use them to implement the logic or something like this, not that they are essential.